File: EaterSpring/src/main/kotlin/xyz/purema/eater/exceptions/NotFoundException.kt
package xyz.purema.eater.exceptions

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import xyz.purema.eater.dtos.BaseResponseDto

class NotFoundException(message: String) : AppException(message) {
    override fun getResponse(): ResponseEntity<BaseResponseDto> = ResponseEntity
        .status(HttpStatus.NOT_FOUND)
        .body(BaseResponseDto(message ?: "Unknown error"))
}

File: EaterNet/Eater/Exceptions/NotFoundException.cs
using Eater.Dtos;
using Microsoft.AspNetCore.Mvc;

namespace Eater.Exceptions
{
    public class NotFoundException : AppException
    {
        public NotFoundException(string message) : base(message)
        {
        }

        public override IActionResult GetResponse()
        {
            return new NotFoundObjectResult(new BaseResponseDto(Message));
        }
    }
}
